### PR TITLE
feat(frontend): replace moment with dayjs

### PR DESCRIPTION
**Pull request by joaoviana**
Relates to: #6129 

### Description:

Replaces `moment` with `dayjs` on the frontend. 

We're already using `dayjs` in some parts of the package, so this replaces all occurrences of `moment`

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

test-cli test-backend test-frontend

### CODE REVIEW

File: packages/frontend/vite.config.ts [hunk 1]
```diff
@@ -46,8 +46,6 @@ export default defineConfig({
                     vega: ['vega', 'vega-lite'],
                     ace: ['ace-builds', 'react-ace/lib'],
                     modules: [
-                        'moment/moment.js',
```
Comment (joaoviana) [inline packages/frontend/vite.config.ts:49]: should `dayjs` be added here? 🤔

Comment (IrakliJani) [reply packages/frontend/vite.config.ts:49]: No need, I think day.js package is tree-shakeable.

File: packages/common/src/utils/formatting.ts [hunk 2]
```diff
@@ -97,11 +98,13 @@ const getTimeFormat = (
     return `YYYY-MM-DD, ${timeFormat} (Z)`;
 };
 
+// TODO: To rename to isDayJsInput once we remove moment usage
 export const isMomentInput = (value: unknown): value is MomentInput =>
```
Comment (IrakliJani) [inline packages/common/src/utils/formatting.ts:102]: little bit afraid of this change as `isMomentInput` is used in backend/common/frontend at the same time. did you have a chance to test all 3 cases?

Comment (joaoviana) [reply packages/common/src/utils/formatting.ts:102]: I've tested the following against 2 render environments: one with and one without this change. 

FE: 
- Tooltips on hover of dates of charts 
- `Filter dashboard to` with dates on 3 different timezones: London, Madrid, and San Francisco
- `Filter chart to` with dates on 3 different timezones: London, Madrid, and San Francisco
- Scheduled deliveries' filters: Change filter of `in the last completed

coming soon with `common` report

Comment (joaoviana) [reply packages/common/src/utils/formatting.ts:102]: The `backend` doesn't use `dayjs` at the moment so this `isMomentInput` change (adding an extra OR) won't be impacted.

But I'm looking at places where `common` is using `dayjs` and see where those functions are called (along with `isMomentInput`

File: packages/frontend/vite.config.ts (hunk 1 above)
Comment (joaoviana) [reply packages/frontend/vite.config.ts:49]: It's not tree shakeable but it's the usage of a core package that is lightweight + plugins that extend functionality https://github.com/iamkun/dayjs/issues/1235#issuecomment-731605523

File: packages/common/src/utils/formatting.ts (hunk 2 above)
Comment (joaoviana) [reply packages/common/src/utils/formatting.ts:102]: Backend: 
`CsvService` uses this for getting the row value for `timestamps` and `dates` here: https://github.com/lightdash/lightdash/blob/feat/frontend-replace-moment-with-dayjs/packages/backend/src/services/CsvService/CsvService.ts#L93 and also uses `formatItemValue`, which also calls `isMomentInput`:

This PR's [render chart table with lots of date dimensions](https://lightdash-pr-9323.onrender.com/projects/3675b69e-8324-4110-bdca-059031aa8da3/tables/events?create_saved_chart_version=%7B%22tableName%22%3A%22events%22%2C%22metricQuery%22%3A%7B%22exploreName%22%3A%22%22%2C%22dimensions%22%3A%5B%22events_timestamp_ltz_raw%22%2C%22events_timestamp_ltz_day%22%2C%22events_timestamp_ntz_raw%22%2C%22events_timestamp_ntz_day%22%2C%22events_timestamp_tz_raw%22%2C%22events_timestamp_tz_millisecond%22%2C%22events_timestamp_tz_minute%22%2C%22events_timestamp_tz_second%22%2C%22events_timestamp_tz_hour%22%2C%22events_timestamp_tz_day%22%5D%2C%22metrics%22%3A%5B%5D%2C%22filters%22%3A%7B%7D%2C%22sorts%22%3A%5B%7B%22fieldId%22%3A%22events_timestamp_tz_day%22%2C%22descending%22%3Atrue%7D%5D%2C%22limit%22%3A500%2C%22tableCalculations%22%3A%5B%5D%2C%22additionalMetrics%22%3A%5B%5D%7D%2C%22tableConfig%22%3A%7B%22columnOrder%22%3A%5B%22events_timestamp_ltz_raw%22%2C%22events_timestamp_ltz_day%22%2C%22events_timestamp_ntz_raw%22%2C%22events_timestamp_ntz_day%22%2C%22events_timestamp_tz_raw%22%2C%22events_timestamp_tz_millisecond%22%2C%22events_timestamp_tz_minute%22%2C%22events_timestamp_tz_second%22%2C%22events_timestamp_tz_hour%22%2C%22events_timestamp_tz_day%22%5D%7D%2C%22chartConfig%22%3A%7B%22type%22%3A%22cartesian%22%2C%22config%22%3A%7B%22layout%22%3A%7B%22xField%22%3A%22events_timestamp_ltz_raw%22%2C%22yField%22%3A%5B%22events_timestamp_ltz_day%22%5D%7D%2C%22eChartsConfig%22%3A%7B%22series%22%3A%5B%7B%22type%22%3A%22bar%22%2C%22yAxisIndex%22%3A0%2C%22encode%22%3A%7B%22xRef%22%3A%7B%22field%22%3A%22events_timestamp_ltz_raw%22%7D%2C%22yRef%22%3A%7B%22field%22%3A%22events_timestamp_ltz_day%22%7D%7D%7D%5D%7D%7D%7D%7D)
Another [render](https://lightdash-pr-9355.onrender.com/projects/3675b69e-8324-4110-bdca-059031aa8da3/tables/events?create_saved_chart_version=%7B%22tableName%22%3A%22events%22%2C%22metricQuery%22%3A%7B%22exploreName%22%3A%22events%22%2C%22dimensions%22%3A%5B%22events_timestamp_ltz_raw%22%2C%22events_timestamp_ltz_day%22%2C%22events_timestamp_ntz_raw%22%2C%22events_timestamp_ntz_day%22%2C%22events_timestamp_tz_raw%22%2C%22events_timestamp_tz_millisecond%22%2C%22events_timestamp_tz_minute%22%2C%22events_timestamp_tz_second%22%2C%22events_timestamp_tz_hour%22%2C%22events_timestamp_tz_day%22%5D%2C%22metrics%22%3A%5B%5D%2C%22filters%22%3A%7B%7D%2C%22sorts%22%3A%5B%7B%22fieldId%22%3A%22events_timestamp_tz_day%22%2C%22descending%22%3Atrue%7D%5D%2C%22limit%22%3A500%2C%22tableCalculations%22%3A%5B%5D%2C%22additionalMetrics%22%3A%5B%5D%7D%2C%22tableConfig%22%3A%7B%22columnOrder%22%3A%5B%22events_timestamp_ltz_raw%22%2C%22events_timestamp_ltz_day%22%2C%22events_timestamp_ntz_raw%22%2C%22events_timestamp_ntz_day%22%2C%22events_timestamp_tz_raw%22%2C%22events_timestamp_tz_millisecond%22%2C%22events_timestamp_tz_minute%22%2C%22events_timestamp_tz_second%22%2C%22events_timestamp_tz_hour%22%2C%22events_timestamp_tz_day%22%5D%7D%2C%22chartConfig%22%3A%7B%22type%22%3A%22cartesian%22%2C%22config%22%3A%7B%22layout%22%3A%7B%22xField%22%3A%22events_timestamp_ltz_raw%22%2C%22yField%22%3A%5B%22events_timestamp_ltz_day%22%5D%7D%2C%22eChartsConfig%22%3A%7B%22series%22%3A%5B%7B%22type%22%3A%22bar%22%2C%22yAxisIndex%22%3A0%2C%22encode%22%3A%7B%22xRef%22%3A%7B%22field%22%3A%22events_timestamp_ltz_raw%22%7D%2C%22yRef%22%3A%7B%22field%22%3A%22events_timestamp_ltz_day%22%7D%7D%7D%5D%7D%7D%7D%7D)

Tested downloading the csv of these results in 2 ways:
- Formatted
- Raw
 and saw no diffs between the two (used online diff check tool)
 
 
 `formatItemValue` here https://github.com/lightdash/lightdash/blob/feat/frontend-replace-moment-with-dayjs/packages/common/src/utils/formatting.ts#L365
 - we have the timezone tests
 - `formatRows` on the BE: https://github.com/lightdash/lightdash/blob/feat/frontend-replace-moment-with-dayjs/packages/common/src/index.ts#L911 - tested this by looking at the rows in table vizualisations + on tooltips and line charts, and dashboard charts with the `filter dashboard to` and dates and consistent between render environments

Tested `formatRows`: 
- On charts
before
Screenshot 2024-03-13 at 17 45 11
after
Screenshot 2024-03-13 at 17 43 52
- On charts in dashboard
before
Screenshot 2024-03-13 at 17 45 36
after
Screenshot 2024-03-13 at 17 45 47
- On underlying data
 
before
Screenshot 2024-03-13 at 17 52 21
after
Screenshot 2024-03-13 at 17 52 28

Comment (joaoviana) [reply packages/common/src/utils/formatting.ts:102]: Also added a simple test here: https://github.com/lightdash/lightdash/pull/9323/commits/87648914f2390d7e94670c4abafee90f94fc0280

File: packages/frontend/vite.config.ts (hunk 1 above)
Comment (IrakliJani) [reply packages/frontend/vite.config.ts:49]: kool!

**Review by IrakliJani — APPROVED** [review]
This looks good! Thorough testing makes me feel safe especially with timezone stuff 🚀

**Comment by IrakliJani** [comment]
btw, im assuming tests are failing because the seeds need to be reset

**Comment by joaoviana** [comment]
> btw, im assuming tests are failing because the seeds need to be reset

will reset Render and re-run, thanks!

**Comment by joaoviana** [comment]
Merging, e2e test for `dashboards.cy.ts` is being fixed here: https://github.com/lightdash/lightdash/pull/9365

**Pull request merged**